scripts/pipeline/orthology_benchmark.py

```diff
@@ -42,46 +42,54 @@
 from ensembl.compara.config import get_species_set_by_name
 
 
-def dump_genomes(species_list: List[str], species_set_name: str, host: str, port: int,
-                 user: str, out_dir: str) -> None:
+def dump_genomes(core_list: List[str], species_set_name: str, host: str, port: int,
+                 out_dir: str, id_type: str) -> None:
     """Dumps canonical peptides of protein-coding genes for a specified list of species.
 
     Peptides are dumped from the latest available core databases to FASTA files
     using `dump_gene_set_from_core.pl`.
 
     Args:
-        species_list: A list of species (genome) names to dump.
+        core_list: A list of core db names to dump.
         species_set_name: Species set (collection) name.
         host: Database host.
         port: Host port.
-        user: Server username.
         out_dir: Directory to place `species_set_name/core_name.fasta` dumps.
+        id_type: Type of stable ids in .fasta header (gene or protein).
 
     Raises:
-        FileExistsError: If directory `out_dir/species_set_name` already exists.
-        sqlalchemy.exc.OperationalError: If `user` cannot read from `host:port`.
-        RuntimeError: If no cores are found for any species in `species_list` (`species_set_name`)
-            on `host:port`.
+        OSError: If creating `out_dir` fails for any reason.
+        RuntimeError: If command to dump a core fails for any reason.
+        ValueError: If `core_list` is empty.
 
     """
-    cores = get_core_names(species_list, host, port, user)
-    dump_cores = list(cores.values())
-
-    if len(dump_cores) == 0:
-        raise RuntimeError(f"No cores found for the species set '{species_set_name}' on the specified host.")
+    if len(core_list) == 0:
+        raise ValueError("No cores to dump.")
 
     dumps_dir = os.path.join(out_dir, species_set_name)
-    os.mkdir(dumps_dir)
+
+    try:
+        os.mkdir(dumps_dir)
+    except OSError as e:
+        raise OSError(f"Failed to create '{dumps_dir}' directory.") from e
 
     script = os.path.join(os.environ["ENSEMBL_ROOT_DIR"], "ensembl-compara", "scripts", "dumps",
                           "dump_gene_set_from_core.pl")
 
-    for core in dump_cores:
+    for core in core_list:
 
         out_file = os.path.join(dumps_dir, f"{core}.fasta")
 
-        subprocess.run([script, "-core-db", core, "-host", host, "-port", str(port),
-                       "-outfile", out_file], capture_output=True, check=True)
+        try:
+            subprocess.run([script, "-core-db", core, "-host", host, "-port", str(port),
+                            "-outfile", out_file, "-id_type", id_type], capture_output=True, check=True)
```

Comment:
Just a small change that I think is required is to either provide a default `id_type` if not provided or change the `subprocess.run` call to not pass `id_type` if not given